src/routes/test/+page.svelte

```diff
@@ -30,6 +31,38 @@
 	import SearchInput from '$lib/components/ui/input/search-input.svelte'
 	import ColorInput from '$lib/components/ui/input/color-input.svelte'
 	import DateInput from '$lib/components/ui/input/date-input.svelte'
+	import Table from '$lib/components/ui/table/table.svelte'
+	import Tr from '$lib/components/ui/table/tr.svelte'
+	import Th from '$lib/components/ui/table/th.svelte'
+	import Td from '$lib/components/ui/table/td.svelte'
+
+	let selected = $state([false, false])
+	let mixed = $state(false)
+	let selectionAll = $state(false)
+	function selectAll() {
+		selectionAll = !selectionAll
+		const allChecked = selected.every((value) => value)
+		if (allChecked) {
+			mixed = false
+			selected = selected.map(() => false)
+		} else {
+			mixed = false
+			selected = selected.map(() => true)
+		}
+	}
+	function toggleSelection(index: number) {
+		selected[index] = !selected[index]
+		if (selected[0] && selected[1]) {
+			mixed = false
+			selectionAll = true
+		} else if (selected[0] || selected[1]) {
+			mixed = true
+			selectionAll = false
+		} else {
+			mixed = false
+			selectionAll = false
+		}
+	}
```

Comment:
Please build this in a generic way, this will likely be used by end-users so it should be able to handle any number of rows.

One idea is to instead of using `mixed` and `selectionAll` here, is to have a `selected` counter which increments/decrement as the checkboxes are selected. The total amount of checkboxes is the length of the checkbox array.